File: src/SIStorage.Service.Contract/Requests/PackageSelectionParameters.cs
namespace SIStorage.Service.Contract.Requests;

/// <summary>
/// Provides packages selection parameters.
/// </summary>
public sealed class PackageSelectionParameters
{
    /// <summary>
    /// Results sort mode.
    /// </summary>
    public PackageSortMode SortMode { get; set; } = PackageSortMode.Name;

    /// <summary>
    /// Results sort direction.
    /// </summary>
    public PackageSortDirection SortDirection { get; set; } = PackageSortDirection.Ascending;

    /// <summary>
    /// "From" pagination parameter.
    /// </summary>
    public int From { get; set; }

    /// <summary>
    /// "Count" pagination parameter.
    /// </summary>
    public int Count { get; set; } = 20;
}

File: web/src/models/PackageSelectionParameters.ts
import PackageSortDirection from './PackageSortDirection';
import PackageSortMode from './PackageSortMode';

/** Provides packages selection parameters. */
export default interface PackageSelectionParameters {
	/** Results sort mode. */
	sortMode?: PackageSortMode;

	/** Results sort direction. */
	sortDirection?: PackageSortDirection;

	/** "From" pagination parameter. */
	from?: number;

	/** "Count" pagination parameter. */
	count?: number;
}